src/keeper.js

```diff
@@ -92,21 +93,22 @@ export default class keeper {
         let owe27 = slice18.mul(auction.price).div(decimals18);
         let tab27 = auction.tab.div(decimals18);
         // adjust covered debt to tab, such that slice better reflects amount of collateral we'd receive
-        if (owe27.gt(tab27)) {
+        if (owe27.gt(tab27) && slice18.gt(auction.lot)) {
           owe27 = tab27;
           slice18 = owe27.div(auction.price.div(decimals18));
         } else if (owe27.lt(tab27) && slice18.lt(auction.lot)) {
           let chost27 = clip._chost.div(decimals18);
           if (tab27.sub(owe27).lt(chost27)) {
             if (tab27.lte(chost27)) {
               // adjust the penultimate take to avoid partial lot on the final take
-              owe27 = tab27 - chost27;
+              owe27 = tab27.sub(chost27);
             } else {
               // adjust to chost
               owe27 = chost27;
             }
+
+            slice18 = owe27.div(auction.price.div(decimals18));
```

Comment:
This fix is unrelated to LP tokens.  I noticed an error when simulating old auctions that caused a failure because the slice got recalculated here.  Since the slice gets set on line `92`, we really only need to recalculate it if we change `owe27` above.  I moved this recalculation into the conditional and it resolved the error.

As a side note, I'm not sure this conditional works as I never hit it in any of my tests.
The if on line `102` clearly checks if `tab27` <= `chost27`, and if that's true, then subtracting `chost27` from `tab27` will leave us with `0` or a negative number.  This is probably wrong, but I don't know what the original author was trying to accomplish, so I'll leave it until it blows up on is.

Reply:
This line was already fixed in c71a3c425830181df3ec6b05342e35caa0c39dcf; perhaps that was merged before your branch was created.